api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -44,6 +55,15 @@ message DiscoveryRequest {
   // which will be explicitly enumerated in resource_names.
   repeated string resource_names = 3;
 
+  // [#not-implemented-hide:]
+  // Alternative to resource_names field that allows specifying cache
+  // keys along with each resource name. If this is populated in the
+  // first request for a resource type on a stream, resource_names is ignored
+  // for all subsequent requests for that resource type on that stream.
+  // Clients that populate this field must be able to handle responses
+  // from the server where resources are wrapped in a Resource message.
```

Comment:
need to add repeated Resource resources in DiscoveryResponse.

Reply:
No.  The idea here is that the `Resource` message will be used in the existing `resources` field in `DiscoveryResponse`.

Reply:
I see.